### PR TITLE
Feature/doaj seal

### PR DESCRIPTION
**Pull request by Steven-Eardley**
Issue #512. This finally passes the tests so I'm fairly confident the underlying code is ready. There are a couple of questions still open about the layout and images, but can be adjusted later if required.

### CODE REVIEW

File: portality/formcontext/forms.py [hunk 1]
```diff
@@ -370,6 +370,11 @@ class Workflow(Form):
                     ' completed your review.'
     )
 
+class Seal(Form):
+    """ Field to set the DOAJ Seal """
+
+    doaj_seal = BooleanField('Set DOAJ Seal', [validators.Optional()], description='description for doaj seal field.', false_values=(BooleanField.false_values + (False,)))
```
Comment (emanuil-tolev) [inline portality/formcontext/forms.py]: "description for doaj seal field." ? I don't recall what "description" is used for, wasn't it the little gray text below fields though?

**Comment by emanuil-tolev** [comment]
Excellent I think. I've not tried it myself since tbh I'm unsure of the criteria, and hence the conditions required for a manual test, but as far as reviewing goes it's tip-top - tests, forms, the xwalks, front-end (toc). Just the one minor q on the form field description, just because it looks like placeholder text, after that will merge!

File: portality/formcontext/forms.py (hunk 1 above)
Comment (Steven-Eardley) [inline portality/formcontext/forms.py]: Ah yes, you've caught one of the placeholders!

**Comment by Steven-Eardley** [comment]
Thanks, will reassign to you when it's ready to merge - I've had more info on the layout changes in the meantime which I'll include in my next commit.

**Comment by Steven-Eardley** [comment]
That's another set of layout changes added

File: portality/templates/formcontext/maned_journal_review.html [hunk 2]
```diff
@@ -95,8 +95,9 @@ <h3 class="heading-aligned-with-fields">Old journal information</h3>
 
         </div>
 
-        {# include the base journal form which contains all the biblio data #}
-        {% include 'formcontext/_base_journal_form.html' %}
+        {# include the base form which contains all the biblio data #}
+        {# we use the base application here because it contains extra information regarding the DOAJ seal #}
```
Comment (emanuil-tolev) [inline portality/templates/formcontext/maned_journal_review.html]: OK I double checked and at the moment the base journal form does not contain anything the base application form doesn't have as well. That could change however.

Also feels weird to include the base app form in the ManEd *journal* review form.

To tackle both of these "smells", can we refactor the seal to be included in a file separate from the base application file? Just thinking about maintainability here, the journal and application files are driven by quite different needs and changes to either may occur without consulting the other. Making the ManEd journal form **not** based on the base journal form feels quite wrong.

Currently we have ```formcontext/_basic_biblio_fields.html```. How about a ```formcontext/_seal.html``` which is then included in ```formcontext/_base_application_form.html``` as well as ```formcontext/maned_journal_review.html```? Just using an include, the same way basic_biblio_fields is included in both files already.

In other words, instead of using the file which already has what we want, we refactor the seal content out of that file into a separate include, and then very explicitly include the seal content where we want it. The explicitness (yes, it's a word) and the composition together make it clear what is (semantically) supposed to be on the screen and we keep our "basic journal form content is shared by all journals forms" hierarchy in place.

**Comment by emanuil-tolev** [comment]
Hiya - looks good, one non-trivial q about template inheritance (but with a suggestion :)). Since this is urgent to be deployed within the next 2 days feel free to ping me at the sign of trouble - e.g. the includes don't quite work the intended way - it's happened several times to me and took a couple of days to get my head around). I think what I've suggested should work straight off the bat in just a few minutes, but don't hesitate to ping me if it doesn't...

**Comment by Steven-Eardley** [comment]
I agree with your changes. I'll get on it now.

**Comment by Steven-Eardley** [comment]
Done

File: portality/migrate/seal/seal.json [hunk 3]
```diff
@@ -0,0 +1,12 @@
+{
```
Comment (emanuil-tolev) [inline portality/migrate/seal/seal.json]: Actually, sorry for q after merge but how does this add the new seal field? I understand this just specifies which types to apply the changes to, but where are the changes actually defined? Or is it a purely new field created upon saving an object (in which case why use a migration)?

Comment (emanuil-tolev) [inline portality/migrate/seal/seal.json]: I just ctrl+F-ed for "migrate" in the PR's code, so apologies if I've missed it. Looking for "seal" is not a very easy endeavour in this set of code for obv reasons :).

Comment (emanuil-tolev) [inline portality/migrate/seal/seal.json]: Need to .save every object so use a migration to regen .index
